core/lib/server/pub-updatedAt-trigger.db.test.ts

```diff
@@ -0,0 +1,147 @@
+import { describe, expect, it } from "vitest";
+
+import type { PubsId } from "db/public";
+import { CoreSchemaType } from "db/public";
+
+import { mockServerCode } from "../__tests__/utils";
+
+const { testDb } = await mockServerCode();
+const { seedCommunity } = await import("~/prisma/seed/seedCommunity");
+
+const { community, pubFields, pubTypes, stages, pubs } = await seedCommunity({
+	community: {
+		name: "test",
+		slug: "test-server-pub",
+	},
+	pubFields: {
+		Title: { schemaName: CoreSchemaType.String },
+		Description: { schemaName: CoreSchemaType.String },
+		"Some relation": { schemaName: CoreSchemaType.String, relation: true },
+	},
+	pubTypes: {
+		"Basic Pub": {
+			Title: { isTitle: true },
+			"Some relation": { isTitle: false },
+		},
+	},
+	stages: {
+		"Stage 1": {},
+	},
+	pubs: [
+		{
+			pubType: "Basic Pub",
+			values: {
+				Title: "Some title",
+			},
+			stage: "Stage 1",
+		},
+		{
+			pubType: "Basic Pub",
+			values: {
+				Title: "Another title",
+			},
+			relatedPubs: {
+				"Some relation": [
+					{
+						value: "test relation value",
+						pub: {
+							pubType: "Basic Pub",
+							values: {
+								Title: "A pub related to another Pub",
+							},
+						},
+					},
+				],
+			},
+		},
+	],
+	users: {},
+});
+
+// this is testing the trigger defined in prisma/migrations/20241126113759_add_pub_values_updated_at_trigger/migration.sql
+describe("updatedAt trigger", () => {
+	it("should update the updatedAt timestamp on a pub whenever a pubvalue is updated, inserted, or deleted", async () => {
+		// we explicitly do not do this inside of a transaction
+		// as otherwise the triggers will not, you know, trigger
+		const { createPubRecursiveNew } = await import("./pub");
+
+		const pub = await createPubRecursiveNew({
+			communityId: community.id,
+			body: {
+				pubTypeId: pubTypes["Basic Pub"].id,
+				values: {
+					[pubFields.Title.slug]: "Test pub",
+				},
+			},
+		});
+
+		const compareUpdatedAt = async () => {
+			const updatedAtBefore = pub.updatedAt;
```

Comment:
Don't these successive `await compareUpdatedAt()` calls need to compare against the most recent `updatedAt` each time rather than the initial one from when the pub is created? Otherwise it only tests that the first pub value change fired the trigger.

So instead of using `const updatedAtBefore = pub.updatedAt`, we would pass that in as a parameter each time:

```ts
const compareUpdatedAt = async (updatedAtBefore: Date) => {
	const newPub = await testDb
		.selectFrom("pubs")
		.select("updatedAt")
		.where("id", "=", pub.id)
		.executeTakeFirstOrThrow();

	expect(newPub.updatedAt.getTime()).toBeGreaterThan(updatedAtBefore.getTime());
};
```

Reply:
shoot yeah you're right!

Reply:
nice, good thinking returning the new date from the compare function!